authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -118,27 +127,372 @@ public List<String> policyResourceDefinesRule() {
+                  return;
+                }
+
+                // Update the policy
+                rangerClient.updatePolicy(policy.getId(), policy);
+              } catch (RangerServiceException e) {
+                LOG.error("Failed to rename the policy {}!", policy);
+                throw new RuntimeException(e);
+              }
+            });
+  }
+
+  /**
+   * IF remove the SCHEMA, need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` <br>
+   * IF remove the TABLE, need to remove these the relevant policies, `{schema}.*`, `{schema}.*.*`
+   * <br>
+   * IF remove the COLUMN, Only need to remove `{schema}.*.*` <br>
+   */
+  @Override
+  protected void doRemoveMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    if (authzMetadataObject.type().equals(SCHEMA)) {
+      doRemoveSchemaMetadataObject(authzMetadataObject);
+    } else if (authzMetadataObject.type().equals(TABLE)) {
+      doRemoveTableMetadataObject(authzMetadataObject);
+    } else if (authzMetadataObject.type().equals(PATH)) {
+      removePolicyByMetadataObject(authzMetadataObject);
+    } else {
+      throw new IllegalArgumentException(
+          "Unsupported authorization metadata object type: " + authzMetadataObject.type());
+    }
+  }
+
+  /**
+   * Remove the SCHEMA, Need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` permissions.
+   */
+  private void doRemoveSchemaMetadataObject(AuthorizationMetadataObject authzMetadataObject) {
+    Preconditions.checkArgument(
+        authzMetadataObject instanceof PathBasedMetadataObject,
+        "The metadata object must be a PathBasedMetadataObject");
+    Preconditions.checkArgument(
+        authzMetadataObject.type() == SCHEMA, "The metadata object type must be SCHEMA");
+    Preconditions.checkArgument(
+        authzMetadataObject.names().size() == 1, "The metadata object names must be 1");
+    if (RangerHelper.RESOURCE_ALL.equals(authzMetadataObject.name())) {
+      // Remove all schema in this catalog
+      NameIdentifier[] catalogs =
+          GravitinoEnv.getInstance().catalogDispatcher().listCatalogs(Namespace.of(metalake));
+      Arrays.asList(catalogs).stream()
+          .forEach(
+              catalog -> {
+                List<String> catalogLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        NameIdentifier.of(catalog.name()), Entity.EntityType.CATALOG);
+                catalogLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject catalogMetadataObject =
+                              new PathBasedMetadataObject(
+                                  metalake, catalog.name(), locationPath, PATH);
+                          doRemoveSchemaMetadataObject(catalogMetadataObject);
+                        });
+              });
```

Comment:
If the metadata object type is `schema`, will we remove all schemas in all catalogs?